journal/src/main/scala/com/github/j5ik2o/akka/persistence/dynamodb/journal/DynamoDBJournal.scala

```diff
@@ -355,38 +363,42 @@ class DynamoDBJournal(config: Config) extends AsyncWriteJournal with ActorLoggin
       .mapAsync(1)(deserializedRepr => Future.fromTry(deserializedRepr))
       .runForeach(recoveryCallback)
       .map(_ => ())
-    future.onComplete {
+    val traced = traceReporter.fold(future)(_.traceJournalAsyncReplayMessages(context)(future))
+    traced.onComplete {
       case Success(_) =>
         metricsReporter.foreach(_.afterJournalAsyncReplayMessages(newContext))
       case Failure(ex) =>
         metricsReporter.foreach(_.errorJournalAsyncReplayMessages(newContext, ex))
     }
-    future
+    traced
   }
 
   override def asyncReadHighestSequenceNr(persistenceId: String, fromSequenceNr: Long): Future[Long] = {
     implicit val ec: ExecutionContext = defaultExecutionContext
     val pid                           = PersistenceId(persistenceId)
-    val context                       = MetricsReporter.newContext(UUID.randomUUID(), pid)
+    val context                       = Context.newContext(UUID.randomUUID(), pid)
     val newContext                    = metricsReporter.fold(context)(_.beforeJournalAsyncReadHighestSequenceNr(context))
+
     def fetchHighestSeqNr(): Future[Long] = {
       journalDao
         .highestSequenceNr(PersistenceId.apply(persistenceId), SequenceNumber(fromSequenceNr)).runWith(Sink.head)
     }
+
     val future = writeInProgress.get(persistenceId) match {
```

Comment:
Change to def.

Reply:
fixed.
https://github.com/j5ik2o/akka-persistence-dynamodb/pull/943/commits/91e2814a4fcf841b337569bf886e735fd5f6d93e#diff-761788fe0403c57a933e6fcf54d2517b298b1a0afcc9e4ef3586ac5b03c3289eR388